src/System.Private.CoreLib/src/System/Threading/ThreadStore.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime;
+
+namespace System.Threading
+{
+    internal static class ThreadStore
```

Comment:
I think this logic can just live in Thread.CoreRT.cs or in StartupHelpers.cs . It is pretty small and a counter like this is not what we usually call thread store.